jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/ToolConfigurable.kt

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.tools
+
+import com.intellij.openapi.fileChooser.FileChooserDescriptorFactory
+import com.intellij.openapi.options.BoundConfigurable
+import com.intellij.openapi.options.SearchableConfigurable
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.layout.applyToComponent
+import com.intellij.ui.layout.panel
+import software.aws.toolkits.jetbrains.ui.ValidatingPanel
+import software.aws.toolkits.resources.message
+import java.nio.file.Path
+
+class ToolConfigurable : BoundConfigurable(message("executableCommon.configurable.title")), SearchableConfigurable {
+    private val settings = ToolSettings.getInstance()
+    private val manager = ToolManager.getInstance()
+    private val panel by lazy {
+        ValidatingPanel(
+            disposable!!,
+            panel {
+                ToolType.EP_NAME.extensionList.forEach { toolType ->
+                    row(toolType.displayName) {
+                        textFieldWithBrowseButton(
+                            getter = { settings.getExecutablePath(toolType) ?: "" },
```

Comment:
why don't we just null out the value instead? is blank string a special case that means something?

Reply:
the getter has to return a `String` it's just for display, the `textFieldWithBrowseButton` is not nullable.